src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesigntimeLicenseContext.cs

```diff
@@ -77,16 +77,13 @@ public override string GetSavedLicenseKey(Type type, Assembly resourceAssembly)
                     // try everything.
                     foreach (Assembly asm in AppDomain.CurrentDomain.GetAssemblies())
                     {
-                        // Though, I could not repro this, we seem to be hitting an AssemblyBuilder
-                        // when walking through all the assemblies in the current app domain. This throws an
-                        // exception on Assembly.CodeBase and we bail out. Catching exceptions here is not a
-                        // bad thing.
-                        if (asm.IsDynamic)
+                        // Assemblies loaded in memory return empty string from Location.
+                        // The code below throws an exception if this happens.
+                        // Catching exceptions here is not a bad thing.
+                        if (asm.Location == string.Empty)
```

Comment:
It would be nice to cache `asm.Location` in a local to avoid calling it twice.

Reply:
Have updated